### PR TITLE
Update scala to 2.12.0-RC1 for travis

### PR DESCRIPTION
**Pull request by TimothyKlim**

### CODE REVIEW

**Comment by xuwei-k** [comment]
```
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: org.specs2#specs2-scalacheck_2.12.0-RC1;3.8.4: not found
```

https://travis-ci.org/json4s/json4s/jobs/159893582#L887

File: project/Dependencies.scala [hunk 1]
```diff
@@ -16,8 +16,8 @@ object Dependencies {
   lazy val commonsCodec = "commons-codec"                %  "commons-codec"        % "1.9"
   lazy val specs        = Def.setting{
     Seq(
-      "org.specs2" %% "specs2-scalacheck" % "3.8.4" % "test" exclude("org.scalacheck", "scalacheck_2.12.0-M4"),
-      "org.scalacheck" %% "scalacheck" % "1.13.1" % "test"
+      "org.specs2" %% "specs2-scalacheck" % "3.8.5" % "test" exclude("org.scalacheck", "scalacheck_2.12.0-RC1"),
```
Comment (xuwei-k) [inline project/Dependencies.scala]: `exclude("org.scalacheck", "scalacheck_2.12.0-RC1")` is unnecessary

**Comment by xuwei-k** [comment]
please update `crossScalaVersions` as well
https://github.com/json4s/json4s/blob/8e8ff7717d9/project/build.scala#L56

**Comment by seratch** [comment]
LGTM, thank you for your contribution 🎉

**Pull request merged**